uwrt_mars_rover_arm/uwrt_mars_rover_arm_hw/include/uwrt_mars_rover_arm_hw/uwrt_mars_rover_arm_hw_actuator.hpp

```diff
@@ -0,0 +1,147 @@
+{
+namespace ActuatorData
+{
+using CommandData = struct CommandData
+{
+  double velocity{};
+};
+
+using StateData = struct StateData
+{
+  double velocity{};
+  double position{};
+  double
+    iq_current{};  // actuator also provide iq_current - this is not needed for transmission however
+};
+
+}  // namespace ActuatorData
+
+namespace JointData
+{
+using CommandData = struct CommandData
+{
+  double velocity{};
+};
+
+using StateData = struct StateData
+{
+  double velocity{};
+  double position{};
+};
+
+}  // namespace JointData
+
+}  // namespace TransmissionData
+
+class ArmActuatorInterface : public hardware_interface::ActuatorInterface
+{
+public:
+  RCLCPP_SHARED_PTR_DEFINITIONS(ArmActuatorInterface)
+
+  UWRT_MARS_ROVER_ARM_HW_PUBLIC
+  std::vector<hardware_interface::StateInterface> export_state_interfaces() override;
+
+  UWRT_MARS_ROVER_ARM_HW_PUBLIC
+  std::vector<hardware_interface::CommandInterface> export_command_interfaces() override;
+
+  UWRT_MARS_ROVER_ARM_HW_PUBLIC
+  hardware_interface::return_type prepare_command_mode_switch(
+    const std::vector<std::string> & /* start_interfaces */,
+    const std::vector<std::string> & /* stop interfaces */) override;
+
+  UWRT_MARS_ROVER_ARM_HW_PUBLIC
+  hardware_interface::return_type read() override;
+
+  UWRT_MARS_ROVER_ARM_HW_PUBLIC
+  hardware_interface::return_type write() override;
+
+  /*
+         * Lifecycle node member functions
+         *
+         */
+  UWRT_MARS_ROVER_ARM_HW_PUBLIC
+  CallbackReturn on_init(
+    const hardware_interface::HardwareInfo & hardware_info) override;  // intialize
+
+  UWRT_MARS_ROVER_ARM_HW_PUBLIC
+  CallbackReturn on_configure(
+    const rclcpp_lifecycle::State & previous_state) override;  // configuration
+
+  UWRT_MARS_ROVER_ARM_HW_PUBLIC
+  CallbackReturn on_activate(const rclcpp_lifecycle::State & previous_state) override;  // start
+
+  UWRT_MARS_ROVER_ARM_HW_PUBLIC
+  CallbackReturn on_deactivate(const rclcpp_lifecycle::State & previous_state) override;  // stop
+
+  UWRT_MARS_ROVER_ARM_HW_PUBLIC
+  CallbackReturn on_shutdown(const rclcpp_lifecycle::State & previous_state)
+    override;  // free up resources before destruction
+
+  UWRT_MARS_ROVER_ARM_HW_PUBLIC
+  CallbackReturn on_cleanup(
+    const rclcpp_lifecycle::State & previous_state) override;  // set node unconfigured state
+
+protected: /*idk we might extend for some reason*/
+  UWRT_MARS_ROVER_ARM_HW_LOCAL
+  bool configure_transmission();
+
+  // ARM Actuator URDF defines these numbers
+  inline static constexpr std::size_t NUM_JOINTS{1};
+  inline static constexpr std::size_t NUM_STATE_INTERFACES{3};
+  inline static constexpr std::size_t NUM_COMMAND_INTERFACES{1};
+
+  // holds actuator reads/writes
+  TransmissionData::ActuatorData::CommandData actuator_commands{};
+  TransmissionData::ActuatorData::StateData actuator_states{};
+  TransmissionData::JointData::CommandData joint_commands{};
+  TransmissionData::JointData::StateData joint_states{};
```

Comment:
Remove transmission stuff from this pr and address seperately

Reply:
can I keep it pretty pretty pretty please .... I'm not using any transmission stuff "really" .... its really just a place holder of how it would be set up ..... I only use the data members of my defined structs to add to the command & state interfaces

Reply:
if its truly unused, then clang-tidy will prevent you from merging it anyways. I would just stash these lines for a future pr